dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-fee-record-payment-groups.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  calculateTotalCurrencyAndAmount,
+  getFeeRecordPaymentEntityGroupReconciliationData,
+  getFeeRecordPaymentEntityGroupStatus,
+} from '../../../../../helpers';
+import { mapPaymentEntityToPayment } from '../../../../../mapping/payment-mapper';
+import { mapFeeRecordEntityToFeeRecord } from '../../../../../mapping/fee-record-mapper';
+import { FeeRecordPaymentGroup } from '../../../../../types/utilisation-reports';
+import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+
+/**
+ * Maps the fee record payment entity groups to the fee record payment groups
+ * @param feeRecordPaymentEntityGroups - The fee record payment entity groups
+ * @returns The fee record payment groups
+ * @throws {Error} If the group has no payments but has multiple fee records
+ */
+export const mapToFeeRecordPaymentGroups = async (feeRecordPaymentEntityGroups: FeeRecordPaymentEntityGroup[]): Promise<FeeRecordPaymentGroup[]> =>
+  await Promise.all(
+    feeRecordPaymentEntityGroups.map(async (group) => {
+      const groupStatus = getFeeRecordPaymentEntityGroupStatus(group);
+      const groupReconciliationData = await getFeeRecordPaymentEntityGroupReconciliationData(group);
+
+      if (group.payments.length === 0) {
```

Comment:
Same above for comparison operators

Reply:
See [here](https://github.com/UK-Export-Finance/dtfs2/pull/3417#discussion_r1746674905)